Source/WTF/wtf/text/WTFString.h

```diff
@@ -524,6 +524,11 @@ inline bool codePointCompareLessThan(const String& a, const String& b)
     return codePointCompare(a.impl(), b.impl()) < 0;
 }
 
+inline bool operator<(const String& lhs, const String& rhs)
```

Comment:
This seems a little too general.  Instead, could we use a comparator in the definition of the type of m_navigations